compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1605,8 +1605,8 @@ fn linker_with_args<'a, B: ArchiveBuilder<'a>>(
     // such:
     //
     // 1. The local object that LLVM just generated
-    // 2. Upstream rust libraries
-    // 3. Local native libraries
+    // 2. Local native libraries
+    // 3. Upstream rust libraries
```

Comment:
There's prose below that needs to be updated to reflect this as well. It currently says:

> Upstream rust libraries are not allowed to depend on our local native libraries as that would violate the structure of the DAG.

Which, while makes sense, doesn't reflect actual implementation nor this list.

Reply:
Well, the prose still describes the intent correctly, it's just the current implementation is leaky. 
I'll add a clarifying comment.